app/src/org/odk/collect/android/activities/FormEntryActivity.java

```diff
@@ -614,16 +521,168 @@ protected void onActivityResult(int requestCode, int resultCode, Intent intent)
         }
     }
 
-    private IntentWidget getPendingIntentWidget() {
-        IntentWidget bestMatch = null;
+    /**
+     * Performs any necessary relocating and scaling of an image coming from either a
+     * SignatureWidget or ImageWidget (capture or choose)
+     *
+     * @param originalImage the image file returned by the image capture or chooser intent
+     * @param shouldScale if false, indicates that the image is from a signature capture, so should
+     *                not attempt to scale
+     * @return a version of the image that is in its final location but has NOT been scaled, for
```

Comment:
I think this return statement is fairly confusing. Can remove info if it makes it less abstract.